middleware/faq.rst

```diff
@@ -436,3 +436,77 @@ What do I do if a query method/command or pipetask graph generation is slow?
 Adding the ``--log-level sqlalchemy.engine=DEBUG`` option to the :any:`butler <lsst.daf.butler-scripts>` or :any:`pipetask <lsst.ctrl.mpexec-script>` command will allow the SQL queries issued by the command to be inspected.
 Similarly, for a slow query method, adding ``logging.getLogger("sqlalchemy.engine").setLevel(logging.DEBUG)`` can help.
 The resulting query logs can be useful for developers and database administrators to determine what, if anything, is going wrong.
+
+.. _middleware_faq_clean_up_runs:
+
+How do I clean up processing runs I don't need anymore?
+=======================================================
+
+.. py:currentmodule:: lsst.daf.butler
+
+Because a data repository stores information on both a filesystem or object store and a SQL database, deleting datasets completely requires using butler commands, even if you know where the associated files are stored on disk.
+
+For processing runs that follow our usual conventions (following them is automatic if you use ``--output`` and don't override ``--output-run`` when running :any:`pipetask <lsst.ctrl.mpexec-script>`), two different collections are created:
+
+- a `~CollectionType.RUN` collection that directly holds your outputs
+- a `~CollectionType.CHAINED` collection that points to that RUN collection as well as all of your input collections.
+
+If you perform multiple processing runs with the same ``--output``, you'll get multiple `~CollectionType.RUN` collections in the same `~CollectionType.CHAINED` collection.
+The `~CollectionType.CHAINED` collection will have the name you passed to ``--output``, and the RUN collections will start with that and end with a timestamp.
+You can see this structure for your own collections with a command like this one:
+
+.. code:: sh
+
+    $ butler query-collections /repo/main --chains=tree u/jbosch/*
+    u/jbosch/DM-30649                                    CHAINED
+      u/jbosch/DM-30649/20210614T191615Z                 RUN
+      HSC/raw/RC2/9813                                   TAGGED
+      HSC/calib/gen2/20180117                            CALIBRATION
+      HSC/calib/DM-28636                                 CALIBRATION
+      HSC/calib/gen2/20180117/unbounded                  RUN
+      HSC/calib/DM-28636/unbounded                       RUN
+      HSC/masks/s18a                                     RUN
+      HSC/fgcmcal/lut/RC2/DM-28636                       RUN
+      refcats/DM-28636                                   RUN
+      skymaps                                            RUN
+    u/jbosch/DM-30649/20210614T191615Z                   RUN
+
+The RUN collections that directly hold the datasets are what we want to remove in order to free up space, but we have to start by deleting the `~CollectionType.CHAINED` collections that hold them first:
+
+.. code:: sh
+
+    $ butler remove-collection /repo/main u/jbosch/DM-30649 --no-confirm
+
+You can drop the ``--no-confirm`` option if you like.
```

Comment:
In the interests of safety (as most people will likely skim this page for recommended commands), I'd recommend replacing the above with:
```
$ butler remove-collections /repo/main u/jbosch/DM-30649
```
and then replacing the sentence below with:
```
You can add the `--no-confirm` option to skip the confirmation prompt if you like.
```